src/audio/kpb.c

```diff
@@ -715,8 +715,11 @@ static int kpb_buffer_data(struct comp_dev *dev,
 
 	tracev_kpb_with_ids(dev, "kpb_buffer_data()");
 
-	if (kpb->state != KPB_STATE_RUN && kpb->state != KPB_STATE_DRAINING)
+	if (kpb->state != KPB_STATE_RUN && kpb->state != KPB_STATE_DRAINING) {
+		trace_kpb_error("kpb_buffer_data() error: wrong state! (current state %d, state log %x)",
+						 kpb->state, kpb->state_log);
 		return PPL_STATUS_PATH_STOP;
+	}
```

Comment:
This part doesn't have anything to do with the introduction of a new state. You can send this as a separate patch. Even separate PR if you want it merged faster.

Reply:
Okay, separate patch seems sane as I did two unrelated things in one patch.